src/librbd/librbd.cc

```diff
@@ -1790,7 +1790,11 @@ namespace librbd {
       return -EINVAL;
     }
 
-    if (mem_is_zero(bl.c_str(), bl.length())) {
+    bool discard_zero = true;
+    if (g_conf != NULL) {
```

Comment:
You cannot reference ```g_conf``` within librbd since it most likely will not be initialized. You instead would access the config via ```ictx->cct->_conf```